File: part-02/src/test/java/eu/luminis/workshop/smallsteps/helpers/TestLegoStockRepository.java
package eu.luminis.workshop.smallsteps.helpers;

import eu.luminis.workshop.smallsteps.logic.appservice.LegoStockRepository;
import eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects.LegoStoreId;
import eu.luminis.workshop.smallsteps.logic.domainservice.state.StockState;
import io.quarkus.test.Mock;

import javax.enterprise.context.ApplicationScoped;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

@Mock
@ApplicationScoped
public class TestLegoStockRepository implements LegoStockRepository {
    private StockState currentState;

    @Override
    public StockState find(LegoStoreId legoStoreId) {
        return this.currentState;
    }

    @Override
    public void save(LegoStoreId legoStoreId, StockState newState) {
        this.currentState = newState;
    }

    public void assertStockState(StockState expected) {
        assertThat(expected.getReserved()).isEqualTo(currentState.getReserved());
    }

    public void setInitialState(StockState initialState) {
        this.currentState = initialState;
    }
}

File: part-02/src/test/kotlin/eu/luminis/workshop/smallsteps/api/helper/TestLegoStockRepository.kt
package eu.luminis.workshop.smallsteps.api.helper

import eu.luminis.workshop.smallsteps.logic.appService.LegoStockRepository
import eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects.LegoStoreId
import eu.luminis.workshop.smallsteps.logic.domainService.state.StockState
import org.assertj.core.api.Assertions

internal class TestLegoStockRepository(initialState: StockState) : LegoStockRepository {
    private var currentState = initialState

    val stockState: StockState get() = currentState

    override fun find(store: LegoStoreId) = currentState

    override fun save(store: LegoStoreId, newState: StockState) {
        currentState = newState
    }

    fun assertStockState(expected: StockState) {
        Assertions.assertThat(stockState).isEqualTo(expected)
    }
}